tests/test_zooniverseCutouts.py

```diff
@@ -0,0 +1,241 @@
+# This file is part of pipe_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import pandas as pd
+import PIL
+import sys
+import unittest
+import tempfile
+
+import lsst.afw.table
+import lsst.geom
+import lsst.meas.base.tests
+from lsst.pex.config import FieldValidationError
+import lsst.utils.tests
+
+from lsst.analysis.ap import zooniverseCutouts
+
+
+class TestZooniverseCutouts(lsst.utils.tests.TestCase):
+    """Test that ZooniverseCutoutsTask generates images and manifest files
+    correctly.
+    """
+    def setUp(self):
+        bbox = lsst.geom.Box2I(lsst.geom.Point2I(0, 0), lsst.geom.Point2I(100, 100))
+        self.centroid = lsst.geom.Point2D(65, 70)
+        dataset = lsst.meas.base.tests.TestDataset(bbox)
+        dataset.addSource(instFlux=1e5, centroid=self.centroid)
+        self.science, self.scienceCat = dataset.realize(noise=1000.0, schema=dataset.makeMinimalSchema())
+        lsst.afw.table.updateSourceCoords(self.science.wcs, self.scienceCat)
+        self.skyCenter = self.scienceCat[0].getCoord()
+        self.template, self.templateCat = dataset.realize(noise=5.0, schema=dataset.makeMinimalSchema())
+        # A simple and incorrect image difference to have something to plot.
+        self.difference = lsst.afw.image.ExposureF(self.science, deep=True)
+        self.difference.image -= self.template.image
+
+    def test_generate_image(self):
+        """Test that we get some kind of image out.
+
+        It's useful to have a person look at the output via:
+            im.show()
+        """
+        cutouts = zooniverseCutouts.ZooniverseCutoutsTask()
+        cutout = cutouts.generate_image(self.science, self.template, self.difference,
+                                        self.skyCenter)
+        im = PIL.Image.open(cutout)
```

Comment:
Does this need a matching `close`?

Reply:
I've switched this to using a context manager now. Much better.